src/analyses/goto_check.cpp

```diff
@@ -1489,20 +1489,16 @@ void goto_checkt::add_guarded_claim(
 
   if(assertions.insert(guarded_expr).second)
   {
-    goto_program_instruction_typet type=
-      enable_assert_to_assume?ASSUME:ASSERT;
-
-    goto_programt::targett t = new_code.add(goto_programt::instructiont(
-      static_cast<const codet &>(get_nil_irep()),
-      source_location,
-      type,
-      std::move(guarded_expr),
-      {}));
+    auto t = new_code.add(
+      enable_assert_to_assume ? goto_programt::make_assumption(
+                                  std::move(guarded_expr), source_location)
+                              : goto_programt::make_assertion(
+                                  std::move(guarded_expr), source_location));
```

Comment:
There is another instance of this pattern (around line 1800) - just search for "ASSUME". Could you please fix this as well?

Reply:
Added.